leptos_dom/src/helpers.rs

```diff
@@ -134,10 +148,95 @@ pub fn set_timeout(cb: impl FnOnce() + 'static, duration: Duration) {
     }
 
     let cb = Closure::once_into_js(Box::new(cb) as Box<dyn FnOnce()>);
-    _ = window().set_timeout_with_callback_and_timeout_and_arguments_0(
-        cb.as_ref().unchecked_ref(),
-        duration.as_millis().try_into().unwrap_throw(),
-    );
+    window()
+        .set_timeout_with_callback_and_timeout_and_arguments_0(
+            cb.as_ref().unchecked_ref(),
+            duration.as_millis().try_into().unwrap_throw(),
+        )
+        .map(TimeoutHandle)
+}
+
+/// Handle that is generated by [set_interval] and can be used to clear the interval.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct TimeoutHandle(i32);
```

Comment:
This struct and its impl could potentially be moved up in this file to be above the `set_timeout` function for consistency with how `set_iterval` has its handle above its function call, which might make it easier for people to find things if these functions and their handles all follow the same pattern.

Reply:
Agreed.